spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,272 @@
-                                                             name: nil)
-          end.not_to change(EquipmentModel, :count)
-        end
-        it { is_expected.to render_template(:new) }
+        it { is_expected.not_to set_flash }
+        it { is_expected.to render_template(:edit) }
       end
     end
-    context 'with non-admin user' do
-      before { sign_in FactoryGirl.create(:user) }
-      it 'should redirect to root' do
-        post :create,
-             equipment_model: FactoryGirl.attributes_for(:equipment_model)
-        expect(response).to redirect_to(root_url)
+    context 'when not admin' do
+      before do
+        mock_user_sign_in
+        put :update, id: 1, equipment_model: { name: 'Model' }
       end
+      it_behaves_like 'redirected request'
     end
   end
 
-  describe 'PUT update' do
-    context 'with admin user' do
-      before { sign_in FactoryGirl.create(:admin) }
-      context 'with valid attributes' do
+  describe 'PUT deactivate' do
+    context 'as admin' do
+      before { mock_user_sign_in(mock_user(:admin)) }
+      shared_examples 'not confirmed' do |flash_type, **opts|
+        let!(:model) { FactoryGirl.build_stubbed(:equipment_model) }
         before do
-          put :update, id: model, equipment_model:
-          FactoryGirl.attributes_for(:equipment_model, name: 'Mod')
-        end
-        it { is_expected.to set_flash }
-        it 'sets @equipment_model to selected model' do
-          expect(assigns(:equipment_model)).to eq(model)
-        end
-        it 'updates attributes' do
-          model.reload
-          expect(model.name).to eq('Mod')
+          allow(EquipmentModel).to receive(:find).with(model.id.to_s)
+            .and_return(model)
+          allow(model).to receive(:destroy)
+          put :deactivate, id: model.id, **opts
         end
+        it { is_expected.to set_flash[flash_type] }
         it { is_expected.to redirect_to(model) }
+        it 'does not deactivate model' do
+          expect(model).not_to have_received(:destroy)
+        end
       end
+      it_behaves_like 'not confirmed', :error
+      it_behaves_like 'not confirmed', :notice, deactivation_cancelled: true
 
-      context 'without valid attributes' do
+      context 'confirmed' do
+        let!(:model) { mock_eq_model(traits: [:findable]) }
         before do
-          put :update, id: model, equipment_model:
-          FactoryGirl.attributes_for(:equipment_model, name: nil)
+          allow(EquipmentModel).to receive(:find).with(model.id.to_s)
+            .and_return(model)
+          request.env['HTTP_REFERER'] = 'where_i_came_from'
+          put :deactivate, id: model.id, deactivation_confirmed: true
         end
-        it { is_expected.not_to set_flash }
-        it 'should not update attributes' do
-          model.reload
-          expect(model.name).not_to be_nil
+        it { is_expected.to set_flash[:notice] }
+        it { is_expected.to redirect_to('where_i_came_from') }
```

Comment:
I believe that this redirect is a part of the controller action - when you deactivate an equipment model using the button (on whatever page you're on), you're supposed to redirect to that page. I guess that in the future this would be handled silently as an API call that would then feed into our React frontend, but that goes back to #1061. Correct me if I'm wrong?